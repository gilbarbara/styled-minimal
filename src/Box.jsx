import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
  alignContent,
  alignItems,
  alignSelf,
  backgroundImage,
  borders,
  borderRadius,
  bottom,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontFamily,
  fontSize,
  fontStyle,
  fontWeight,
  height,
  justifyContent,
  left,
  lineHeight,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  order,
  position,
  ratio,
  right,
  space,
  textAlign,
  top,
  width,
  zIndex,
} from 'styled-system';

import { textTransform } from './utils/system';

export const Box = styled.div(
  {
    boxSizing: 'border-box',
  },
  alignContent,
  alignItems,
  alignSelf,
  backgroundImage,
  borders,
  borderRadius,
  bottom,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontFamily,
  fontSize,
  fontStyle,
  fontWeight,
  height,
  justifyContent,
  left,
  lineHeight,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  order,
  position,
  ratio,
  right,
  space,
  textAlign,
  textTransform,
  top,
  width,
  zIndex
);

Box.displayName = 'Box';

Box.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node,
  textTransform: PropTypes.string,
  ...alignContent.propTypes,
  ...alignItems.propTypes,
  ...alignSelf.propTypes,
  ...backgroundImage.propTypes,
  ...borders.propTypes,
  ...borderRadius.propTypes,
  ...bottom.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...flex.propTypes,
  ...flexBasis.propTypes,
  ...flexDirection.propTypes,
  ...flexWrap.propTypes,
  ...fontFamily.propTypes,
  ...fontSize.propTypes,
  ...fontWeight.propTypes,
  ...height.propTypes,
  ...justifyContent.propTypes,
  ...left.propTypes,
  ...maxHeight.propTypes,
  ...maxWidth.propTypes,
  ...minHeight.propTypes,
  ...minWidth.propTypes,
  ...order.propTypes,
  ...position.propTypes,
  ...right.propTypes,
  ...ratio.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...top.propTypes,
  ...width.propTypes,
  ...zIndex.propTypes,
};

export default Box;
