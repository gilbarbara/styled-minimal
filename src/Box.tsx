import * as React from 'react';
import styled from 'styled-components';

import { baseStyles, sx, system } from './utils/system';

import { BoxProps } from './types';

export const StyledBox = styled('div')<BoxProps>(baseStyles, sx, system);

export const Box = React.forwardRef<HTMLDivElement, BoxProps>((props, ref) => (
  <StyledBox ref={ref} {...props} />
));
