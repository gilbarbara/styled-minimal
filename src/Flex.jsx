import React from 'react';
import PropTypes from 'prop-types';
import system from 'system-components';
import { systemGroups } from './utils/system';

export const StyledFlex = system({
  display: 'flex',
  width: '100%',
}, ...systemGroups('flex', 'layout', 'positioning'));

const Flex = ({ children, ...props }) => (
  <StyledFlex {...props}>{children}</StyledFlex>
);

Flex.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Flex;
