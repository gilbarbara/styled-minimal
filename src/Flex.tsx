import * as React from 'react';
import styled from 'styled-components';

import { baseStyles, sx, system } from './utils/system';

import { FlexProps } from './types';

export const StyledFlex = styled('div')<FlexProps>(
  {
    ...baseStyles,
    display: 'flex',
  },
  sx,
  system,
);

export const Flex = React.forwardRef<HTMLDivElement, FlexProps>((props, ref) => (
  <StyledFlex ref={ref} {...props} />
));
