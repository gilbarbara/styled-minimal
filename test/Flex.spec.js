import React from 'react';
import { mount } from 'enzyme';

import Box from '../src/Box';
import Flex from '../src/Flex';

describe('Flex', () => {
  const wrapper = mount(
    <Flex>
      <Box>Left</Box>
      <Box>Right</Box>
    </Flex>
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
