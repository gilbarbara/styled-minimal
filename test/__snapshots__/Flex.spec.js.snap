// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should render properly 1`] = `
.c1 {
  box-sizing: border-box;
}

.c0 {
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<Flex
  alignItems="center"
>
  <div
    className="c0"
  >
    <Box>
      <div
        className="c1"
      >
        Left
      </div>
    </Box>
    <Box>
      <div
        className="c1"
      >
        Right
      </div>
    </Box>
  </div>
</Flex>
`;
